public/components/CampaignAnalytics/Analytics/DateRangeEditor.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+class DateRangeEditor extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: DateRangeEditor.toDate(this.props.campaign.startDate),
+      endDate: DateRangeEditor.toDate(this.props.campaign.endDate)
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    if (nextState !== this.state) {
+      this.props.onChange(nextState);
+    }
+  }
+
+  handleChange({target}) {
+    this.setState({
+      [target.name]: target.value
+    });
+  }
+
+  static zeroPad(num) {
+    return ('0' + num).slice(-2)
```

Comment:
I recognise this time-formatting snippet from stackoverflow. a classic!

Reply:
The source of all my knowledge.  😄